index.cjs

```diff
@@ -165,6 +165,24 @@ module.exports = {
     // Warn on restricted syntax
     'no-restricted-syntax': [
       'warn',
+      {
+        selector:
+          "ExpressionStatement CallExpression[callee.object.name='document'][callee.property.name='querySelector']",
+        message: `Using document.querySelector() can give unexpected results, especially if multiple elements match the query (for example, when the component is used more than once); as a piece of advice, use State or Refs instead
+        https://github.com/reactjs/reactjs.org/issues/4626#issuecomment-1117535930`,
+      },
+      {
+        selector:
+          "ExpressionStatement CallExpression[callee.object.name='document'][callee.property.name='querySelectorAll']",
+        message: `Using document.querySelectorAll() can give unexpected results, especially if multiple elements match the query (for example, when the component is used more than once); as a piece of advice, use State or Refs instead
+        https://github.com/reactjs/reactjs.org/issues/4626#issuecomment-1117535930`,
```

Comment:
Can you add a comment about why we can't use Markdown links yet (with all the links to the open issues in VS Code / VS Code extension)?

Here are some issues that we both found, not sure if they are all relevant (or if there are more relevant links which are not in this list):

- https://github.com/microsoft/vscode/issues/11847
- https://github.com/microsoft/vscode-eslint/issues/418
- https://github.com/microsoft/vscode/issues/54272

Reply:
https://github.com/upleveled/eslint-config-upleveled/pull/105/commits/085d47f972cff939f57da74420ddcd2e8f55f758

Reply:
https://github.com/upleveled/eslint-config-upleveled/issues/123